kotlin/workflow-ui-android/src/main/java/com/squareup/workflow/ui/ViewRegistry.kt

```diff
@@ -83,6 +83,36 @@ class ViewRegistry private constructor(
           }
   )
 
+  /**
+   * Updates the given view if it [can show][View.canShowRendering] the given [rendering].
+   * If not, replaces it with one that can.
+   *
+   * Returns the view that is now showing the rendering. If the original view was in a parent
+   * and a new view was required for [rendering], the new view replaces the original in
+   * its parent.
+   */
+  fun <RenderingT : Any> updateView(
+    view: View,
+    rendering: RenderingT
+  ): View {
+    if (view.canShowRendering(rendering)) {
+      view.showRendering(rendering)
+      return view
+    }
+
+    return when (val parent = view.parent) {
+      is ViewGroup -> buildView(rendering, parent).also { newView ->
+        val index = parent.indexOfChild(view)
+        parent.removeView(view)
+
+        view.layoutParams
+            ?.let { parent.addView(newView, index, it) }
+            ?: run { parent.addView(newView, index) }
```

Comment:
This is cool, but almost feels too magic… don't have any better ideas atm though.

Reply:
Yep, just looked at the impl in `ViewStubCompat`, looks very similar.

Reply:
Yes, I took the code from ViewStub. But no, we can't just use ViewStub, if that's where you're going. It's single shot, and it's strictly related to layouts.

Reply:
Didn't mean use it, just use it as the placeholder since it's semantically very similar (and already zero-sized). But don't feel too strongly.